File: flow-client/src/main/java/com/vaadin/client/flow/nodefeature/MapPropertyChangeEvent.java
/*
 * Copyright (C) 2000-2024 Vaadin Ltd
 *
 * This program is available under Vaadin Commercial License and Service Terms.
 *
 * See <https://vaadin.com/commercial-license-and-service-terms> for the full
 * license.
 */
package com.vaadin.client.flow.nodefeature;

import com.vaadin.client.flow.reactive.ReactiveValueChangeEvent;

/**
 * Event fired when the value of a map property changes.
 *
 * @author Vaadin Ltd
 * @since 1.0
 */
public class MapPropertyChangeEvent extends ReactiveValueChangeEvent {

    private Object oldValue;
    private Object newValue;

    /**
     * Creates a new map property change event.
     *
     * @param source
     *            the changed map property
     * @param oldValue
     *            the old value
     * @param newValue
     *            the new value
     */
    public MapPropertyChangeEvent(MapProperty source, Object oldValue,
            Object newValue) {
        super(source);
        this.oldValue = oldValue;
        this.newValue = newValue;
    }

    /**
     * Gets the old property value.
     *
     * @return the old value
     */
    public Object getOldValue() {
        return oldValue;
    }

    /**
     * Gets the new property value.
     *
     * @return the new value
     */
    public Object getNewValue() {
        return newValue;
    }

    @Override
    public MapProperty getSource() {
        return (MapProperty) super.getSource();
    }

}

File: flow-client/src/main/frontend/internal/client/flow/nodefeature/MapPropertyChangeEvent.ts
/*
 * Copyright 2000-2026 Vaadin Ltd.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */

// TypeScript port of com.vaadin.client.flow.nodefeature.MapPropertyChangeEvent.

import { ReactiveValueChangeEvent } from '../reactive/ReactiveValueChangeEvent';
import type { MapProperty } from './MapProperty';

/**
 * Event fired when the value of a map property changes.
 */
export class MapPropertyChangeEvent extends ReactiveValueChangeEvent {
  readonly #oldValue: unknown;

  readonly #newValue: unknown;

  /**
   * Creates a new map property change event.
   *
   * @param source - the changed map property
   * @param oldValue - the old value
   * @param newValue - the new value
   */
  constructor(source: MapProperty, oldValue: unknown, newValue: unknown) {
    super(source);
    this.#oldValue = oldValue;
    this.#newValue = newValue;
  }

  override getSource(): MapProperty {
    return super.getSource() as MapProperty;
  }

  /**
   * Gets the old property value.
   *
   * @returns the old value
   */
  getOldValue(): unknown {
    return this.#oldValue;
  }

  /**
   * Gets the new property value.
   *
   * @returns the new value
   */
  getNewValue(): unknown {
    return this.#newValue;
  }
}
